### ISSUE TITLE
TSLint does not respect the underscore pattern for unused parameters

### ISSUE THREAD

**Issue by pronebird** [palantir/tslint#4505]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.x
- __Running TSLint via__: CLI

#### Actual behavior

TSLint does not respect the underscore pattern for unused parameters.

p.s: The code below is stripped down a little to demonstrate the issue.

##### Take 1

Use unused parameters:

```ts
ipcRenderer.on(event, (unusedEvent: Electron.Event, newState: T) => fn(newState));
```

Get punched by TypeScript:

>error TS6133: 'unusedEvent' is declared but its value is never read.

##### Take 2

Underscore the unused parameter

```ts
ipcRenderer.on(event, (_unusedEvent: Electron.Event, newState: T) => fn(newState));
```

Get praised by TypeScript:

>$ tsc
>✨  Done in 8.15s.

##### Take 3

Try linting the code that TypeScript is happy about:

>ERROR: 297:28  variable-name                     variable name must be in lowerCamelCase, PascalCase or UPPER_CASE

Very cool.

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:latest",
    "tslint-config-prettier"
  ],
  "rules": {
    "max-classes-per-file": false,
    "object-literal-sort-keys": false
  }
}
```

#### Expected behaviour

TSLint should respect the underscore pattern for unused parameters.

**Comment by JoshuaKGoldberg** [comment]
Hi @pronebird! Have you tried using the `allow-leading-underscore` option? https://palantir.github.io/tslint/rules/variable-name/

**Comment by pronebird** [comment]
@JoshuaKGoldberg yep and it works, but I would like this pattern to be allowed only for unused parameters only. I think ti makes sense because it's more of an exception.

**Comment by JoshuaKGoldberg** [comment]
Gotcha (and agreed). This is a duplicate of #3442, then. +1 for allowing this!

**Comment by pronebird** [comment]
Ah right, thanks! 👍
